From 9fb9179a3ab0e3ee34154b5f9867434873a66b7c Mon Sep 17 00:00:00 2001
From: Mihai Serban <mihai.serban@nxp.com>
Date: Thu, 8 Jun 2017 15:48:50 +0300
Subject: [PATCH 1873/5242] MLK-15039: ASoC: fsl_esai: Fix channels swap when
 recording 3 channels audio

commit  bc19a18543e75a7b6dfbb46a09b6539546447206 from
https://source.codeaurora.org/external/imx/linux-imx.git

The change introduced by commit 00c174b3b28a
("MLK-14525: ASoC: fsl_esai: channel swap issue in 3 channels or 5 channels")
is no longer valid after improvements added with commit c35bc6ae5c48
("MLK-14778: ASoC: fsl: imx-cs42888: Improve support for odd number of channels")

Because we use TDM instead of I2S for 3,5 and 7 channels we must
initialize ESAI with the actual number of channels. There is no need
to count the additional channel required when I2S was used.

Signed-off-by: Mihai Serban <mihai.serban@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_esai.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 1fb1ddd..361d15f 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -838,7 +838,7 @@ static int fsl_esai_trigger(struct snd_pcm_substream *substream, int cmd,
 				   ESAI_xFCR_xFEN_MASK, ESAI_xFCR_xFEN);
 
 		/* Write initial words reqiured by ESAI as normal procedure */
-		for (i = 0; tx && i < pins * esai_priv->slots; i++)
+		for (i = 0; tx && i < channels; i++)
 			regmap_write(esai_priv->regmap, REG_ESAI_ETDR, 0x0);
 
 		regmap_update_bits(esai_priv->regmap, REG_ESAI_xCR(tx),
-- 
1.7.9.5

